lib.php

```diff
@@ -5723,6 +5723,13 @@ function hsuforum_print_discussion($course, $cm, $forum, $discussion, $post, $ca
         $reply = $canreply;
     }
 
+    // Add a check that user should be able to reply ...
+    // ...to posts that has passed expected completion date...
+    // ...if the expected completion date is enabled.
+    if ($cm->completionexpected && $cm->completionexpected < time()) {
+        $reply = false;
+    }
```

Comment:
This will completely lock replies. Is this the expected functionality? Like are there use cases where for example the academic team might want to post something after the forum should be completed?